packages/react-components/src/localization/LocalizationProvider.tsx

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { createContext, useState, useMemo, useContext, useCallback, useEffect } from 'react';
+import { defaultLocaleStringsLoader, locales } from './loadLocaleStrings';
+import defaultStrings from './translated/en-US.json';
+
+const LOCALE_CACHE_KEY = 'AzureCommunicationUI_Locale';
+
+export interface ILocale {
+  locale: string;
+  englishName: string;
+  displayName: string;
+  rtl: boolean;
+}
+
+/**
+ * Collection of ILocale
+ */
+export type ILocaleCollection = Record<string, ILocale>;
+
+export type ILocaleKeys = Record<string, string>;
+
+export interface ILocaleContext {
+  locale: ILocale;
+  locales: ILocaleCollection;
+  strings: ILocaleKeys;
+  setLocale: (locale: string, forceReload?: boolean) => void;
+}
+
+const defaultLocaleContext: ILocaleContext = {
+  locale: locales['en-US'],
+  locales: locales,
+  strings: defaultStrings,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  setLocale: (locale: string, forceReload?: boolean) => {}
+};
+
+export const LocaleContext = createContext<ILocaleContext>(defaultLocaleContext);
+
+export type LocalizationProviderProps = {
+  initialLocale: string;
+  locales: ILocaleCollection;
+  localeStringsLoader?: (locale: string) => Promise<ILocaleKeys>;
+  storage?: Storage;
+  children: React.ReactNode;
+};
+
+export const LocalizationProvider = (props: LocalizationProviderProps): JSX.Element => {
+  const { children, initialLocale, locales, storage, localeStringsLoader } = props;
+
+  useEffect(() => {
+    if (storage) {
+      const cachedLocale = storage.getItem(LOCALE_CACHE_KEY);
+      if (cachedLocale) {
+        setLocale(cachedLocale);
+      } else {
+        storage.setItem(LOCALE_CACHE_KEY, locale.locale);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!locales[initialLocale]) {
+    throw new Error('Initial locale assigned is not in locale collection assigned.');
+  }
+
+  const [locale, _setLocale] = useState<ILocale>(locales[initialLocale]);
+  const [strings, setStrings] = useState<ILocaleKeys>({});
+
+  const loadLocaleStrings = localeStringsLoader ?? defaultLocaleStringsLoader;
+
+  const setLocale = useCallback(
+    async (locale: string, forceReload?: boolean) => {
+      const loc = locales[locale];
+
+      if (loc) {
+        if (storage) {
+          storage.setItem(LOCALE_CACHE_KEY, loc.locale);
+        }
+
+        if (forceReload) {
+          window.location.reload();
+        } else if (loadLocaleStrings) {
+          const localeData = await loadLocaleStrings(loc.locale);
+          setStrings(localeData);
+          _setLocale(loc);
+
+          document.documentElement.lang = loc.locale;
+          document.documentElement.dir = loc.rtl ? 'rtl' : 'ltr';
```

Comment:
RTl vs LTR is normally set at the HTML level of a page: https://www.w3schools.com/TAgs/att_dir.asp#:~:text=The%20dir%20attribute%20specifies%20the%20text%20direction%20of,and%20can%20be%20used%20on%20any%20HTML%20element.

Looks like fluent also grabs it from here: https://developer.microsoft.com/en-us/fluentui#/styles/web/localization

Reply:
Right. I could be wrong but doesn't document.documentElement get the root element of a page which is <html> for an HTML page?

Reply:
Yep you're right, I was thinking the user would set `rtl` and `lang` in their html markup and we would pick it up from there instead of defining it per locale and overwriting whatever was set already

Reply:
(happy if we should be setting it based on chosen locale, but just want to check)